src/test/java/com/google/research/bleth/simulator/GlobalResolverTest.java

```diff
@@ -21,211 +21,211 @@ public void estimatedBoardInitializedEmpty() {
-                                                                ImmutableList.of(beacon));
+        GlobalResolver resolver = GlobalResolver.create(/* rowsNum= */ 2,
+                                                        /* colsNum= */ 2,
+                                                        ImmutableList.of(beacon));
 
         resolver.estimate();
 
-        assertThat(isBoardEmpty(resolver.getBoard(), 2, 2));
+        assertThat(resolver.getBoard().agentsOnBoard()).isEmpty();
     }
 
     @Test
-    public void firstEstimateAfterReceivingZeroTransmissions_EstimatedBoardIsEmpty() {
+    public void firstEstimateAfterReceivingZeroTransmissions_estimatedBoardIsEmpty() {
         RealBoard realboard = new RealBoard(2, 2);
         Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
-        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 2,
-                                                                /* colsNum= */ 2,
-                                                                ImmutableList.of(beacon));
+        GlobalResolver resolver = GlobalResolver.create(/* rowsNum= */ 2,
+                                                        /* colsNum= */ 2,
+                                                        ImmutableList.of(beacon));
 
         resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of());
         resolver.estimate();
 
-        assertThat(isBoardEmpty(resolver.getBoard(), 2, 2));
+        assertThat(resolver.getBoard().agentsOnBoard()).isEmpty();
     }
 
     @Test
-    public void firstEstimateWithInformationFromObserverOnSameLocationAsBeacon_UpdateEstimatedBoardAccordingToItsLocation() {
+    public void firstEstimateWithInformationFromObserverOnSameLocationAsBeacon_updateEstimatedBoardAccordingToItsLocation() {
         RealBoard realboard = new RealBoard(2, 2);
         Beacon beacon = createRandomBeaconOnLocation(ZERO_ON_ZERO_COORDINATE, realboard);
-        GlobalResolver resolver = GlobalResolver.createResolver(/* rowsNum= */ 2,
-                                                                /* colsNum= */ 2,
-                                                                ImmutableList.of(beacon));
+        GlobalResolver resolver = GlobalResolver.create(/* rowsNum= */ 2,
+                                                        /* colsNum= */ 2,
+                                                        ImmutableList.of(beacon));
 
         resolver.receiveInformation(ZERO_ON_ZERO_COORDINATE, ImmutableList.of(beacon.transmit()));
         resolver.estimate();
 
-        assertThat(resolver.getBoard().getAgentsOnLocation(ZERO_ON_ZERO_COORDINATE)).containsExactly(beacon);
-        assertThat(resolver.getBoard().getAgentsOnLocation(ONE_ON_ONE_COORDINATE)).isEmpty();
+        assertThat(resolver.getBoard().agentsOnBoard().get(ZERO_ON_ZERO_COORDINATE)).containsExactly(beacon);
```

Comment:
Now that you have `agentsOnBoard`, you could simplify both assertions here to a single

```
assertThat(resolver.getBoard().agentsOnBoard()
    .containsExactly(ImmutableListMultimap.of(ZERO_ON_ZERO_COORDINATE, beacon));
```

Reply:
(similar comment apply to the other test cases; many can be simplified as well)